src/analysis/calculus/fderiv.lean

```diff
@@ -1223,30 +1223,30 @@ section fst
 
 variables {f₂ : E → F × G} {f₂' : E →L[𝕜] F × G} {p : E × F}
 
-lemma has_strict_fderiv_at_fst : has_strict_fderiv_at prod.fst (fst 𝕜 E F) p :=
+lemma has_strict_fderiv_at_fst : has_strict_fderiv_at (@prod.fst E F) (fst 𝕜 E F) p :=
```

Comment:
Do you know why this has become necessary? It's unfortunate. :-(

Reply:
I don't know.. as I said in the other comment it seems Lean as lost the ability the implicit variables sometimes

Reply:
I changed the priority for the instance on the product to `@[priority 100]`, and now
```
lemma has_strict_fderiv_at_fst : has_strict_fderiv_at (fst 𝕜 E F) (fst 𝕜 E F) p :=
(fst 𝕜 E F).has_strict_fderiv_at
```
works. The original version still doesn't work, but this is nicer...